### PR TITLE
CLN: clean setup.py, remove py.test command

### PR DESCRIPTION
**Pull request by fangchenli**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,49 +1,11 @@
-#!/usr/bin/env python
```
Comment (datapythonista) [inline setup.py:1]: I'd leave this.

File: setup.py [hunk 2]
```diff
@@ -1,49 +1,11 @@
...
-            test_args += self.pytest_args.split()
-        if self.coverage:
-            test_args += ['--cov', os.path.abspath('asv')]
-        errno = pytest.main(test_args)
-        if errno != 0:
-            raise DistutilsError("Tests failed")
+from setuptools import Extension, setup
+from setuptools.command.build_ext import build_ext
+from setuptools.command.sdist import sdist
+from setuptools.errors import CCompilerError, ExecError, PlatformError
```
Comment (datapythonista) [inline setup.py:8]: Looks like `CCompilerError` can't be imported, not sure what's the new exception.

**Review by datapythonista — COMMENTED** [review]
Nice clean up, just couple of small things.

File: setup.py (hunk 2 above)
Comment (fangchenli) [reply setup.py:8]: setuptools vendored those exceptions from distutils. And it's only available on very recent releases. I'll revert this.

**Comment by datapythonista** [comment]
Thanks for the clean up @fangchenli

**Pull request merged**